src/NexusMods.Abstractions.Downloads/DownloadInfo.cs

```diff
@@ -79,10 +90,52 @@ public class DownloadInfo : ReactiveObject
     /// We store the EntityId rather than NexusModsFileMetadata.ReadOnly to avoid circular dependency issues.
     /// The full FileMetadata can be retrieved from the database using this ID.
     /// </remarks>
-    [Reactive] public required EntityId FileMetadataId { get; set; }
+    public IReadOnlyBindableReactiveProperty<EntityId> FileMetadataId => _fileMetadataId;
     
     /// <summary>
     /// Internal subscription management for reactive updates.
     /// </summary>
-    internal CompositeDisposable? Subscriptions { get; set; }
+    internal System.Reactive.Disposables.CompositeDisposable? Subscriptions { get; set; }
+
+    // Internal mutation methods for DownloadsService
+    internal void SetName(string value) => _name.Value = value;
+    internal void SetGameId(GameId value) => _gameId.Value = value;
+    internal void SetFileSize(Size value) => _fileSize.Value = value;
+    internal void SetProgress(Percent value) => _progress.Value = value;
+    internal void SetDownloadedBytes(Size value) => _downloadedBytes.Value = value;
+    internal void SetTransferRate(Size value) => _transferRate.Value = value;
+    internal void SetStatus(JobStatus value) => _status.Value = value;
+    internal void SetDownloadPageUri(Optional<Uri> value) => _downloadPageUri.Value = value;
+    internal void SetCompletedAt(Optional<DateTimeOffset> value) => _completedAt.Value = value;
+    internal void SetFileMetadataId(EntityId value) => _fileMetadataId.Value = value;
```

Comment:
If the state is supposed to be mutable from the outside, we could just expose the `BindableReactiveProperty`s directly. 
The previous [Reactive] properties were public so, if we don't care that much we can skip a lot of boilerplate here.

Reply:
The state's only meant to be mutable by the `DownloadsService` which is the source of truth that populates `DownloadInfo`.

I had to do `internal` and `InternalsVisibleTo` however, because `DownloadInfo` is in a separate `Abstractions` project.